haystack/preview/components/converters/pypdf.py

```diff
@@ -42,14 +47,19 @@ class PyPDFToDocument:
     A default text extraction converter is used if no custom converter is provided.
     """
 
-    def __init__(self, converter: Optional[PyPDFConverter] = None):
+    def __init__(self, converter_name: str = "default"):
         """
         Initializes the PyPDFToDocument component with an optional custom converter.
-        :param converter: A converter instance that adheres to the PyPDFConverter protocol.
-                          If None, the DefaultConverter is used.
+        :param converter_name: A converter name that is registered in the CONVERTER_REGISTRY.
+            Defaults to 'default'.
         """
         pypdf_import.check()
-        self.converter: PyPDFConverter = converter or DefaultConverter()
+        self.converter_name = converter_name
+        self._converter: PyPDFConverter = CONVERTERS_REGISTRY[converter_name]
```

Comment:
Let's add a check here to make sure it's there

Reply:
And check type as we use string now

Reply:
And throw some error I suppose

Reply:
I added the checks and some fresh tests...

Reply:
> And check type as we use string now

I think catching the `KeyError` is sufficient.